src/main/java/org/interledger/ilqp/QuoteLiquidityResponse.java

```diff
@@ -48,5 +48,148 @@
    * @return An instance of {@link ZonedDateTime}.
    */
   ZonedDateTime getExpiresAt();
+  
+  /**
+   * A builder for instances of {@link QuoteLiquidityResponse}.
+   */
+  class Builder {
+    private LiquidityCurve liquidityCurve;
+    private InterledgerAddress appliesTo;
+    private Duration sourceHoldDuration;
+    private ZonedDateTime expiresAt;
+    
+    /**
+     * Set the liquidity curve into this builder.
+     *
+     * @param liquidityCurve To An instance of {@link LiquidityCurve}.
+     */
+    public Builder liquidityCurve(final LiquidityCurve liquidityCurve) {
+      this.liquidityCurve = Objects.requireNonNull(liquidityCurve);
```

Comment:
It's a minor style nit, but I prefer to perform null-checks as early as possible (like in these setter methods in the builder) because using null here is a bug - a developer should never null out a field in the builder because null is never valid. Conversely, general validation logic _should_ be in the build method because we don't want to validate potentially in-flight mutable values of the builder until 'build' is called, and then we do want to validate.

Al that to say, I vote for leaving the null checks here.